test/integration/job/job_test.go

```diff
@@ -399,7 +400,9 @@ func TestJobPodFailurePolicyWithFailedPodDeletedDuringControllerRestart(t *testi
 		Ready:  pointer.Int32(0),
 	})
 
-	jobPods, err := getJobPods(ctx, t, cs, jobObj)
+	jobPods, err := getJobPods(ctx, t, cs, jobObj, func(s v1.PodStatus) bool {
+		return (s.Phase == v1.PodPending || s.Phase == v1.PodRunning)
```

Comment:
why this change?

Reply:
In master [code](https://github.com/kubernetes/kubernetes/blob/b99fe0d5b9896dd3fe9a2c1bc3b399a18ad080d2/test/integration/job/job_test.go#LL1689-L1704), we have `getJobPods` which returns all the pods controlled by the given job that is in a pending or running state. For the new test case that we are introducing, we required exactly the same function; the only difference being all pods irrespective of their states should be returned.

So, modified `getJobPods` to include a function argument that evaluates the provided condition. The old test that used `getJobPods` pass a lambda that returns true when the pod's phase is in pending or running. In our newly added test, the lambda simply returns true irrespective of the pod's phase.

The change that we see here is for the existing test case.

Reply:
I see. It would be better to add a function: `getActiveJobPods` to improve readability. But that's a nit.